libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -625,6 +626,31 @@ class MapboxTripSessionTest {
         coVerify(exactly = 1) { navigator.setRoute(null) }
     }
 
+    @Test
+    fun sessionStopWaitsForCompleteOfSetRouteJob() = coroutineRule.runBlockingTest {
+        val routeInitInfo = mockk<RouteInitInfo>()
+        val routeAlert = mockk<RouteAlert>()
+        every { routeInitInfo.routeAlerts } returns listOf(routeAlert)
+        coEvery { navigator.setRoute(any()) } coAnswers {
+            // but we need to check weather we wait until setRouteJob finishes when we stop the session.
+            // without the delay coroutine might be finished to fast and our test will be useless
+            // (it will pass even if we don't wait for a coroutine)
+            delay(10)
+            routeInitInfo
+        }
+
+        val routeAlertsObserver = mockk<RouteAlertsObserver>(relaxUnitFun = true)
+        tripSession.registerRouteAlertsObserver(routeAlertsObserver)
+
+        tripSession.start()
+        tripSession.route = null
+        tripSession.stop()
+
+        tripSession.stopSessionJob?.join()
+
+        verify(exactly = 2) { routeAlertsObserver.onNewRouteAlerts(any()) }
```

Comment:
How does this verify the correctness of the situation?

Should we be rather checking something like
`verify { navigator.setRoutes(null) }`

Reply:
@LukasPaczos 
when we set `route = null`
we update `routeAlerts` twice. The second time inside a coroutine after navigator's response.
it's a workaround to check that `coroutine` inside `set route` was finised before we call `stop()`.
I've tried to find a better solution and failed ((
we can't check something like `setRouteJob.isCompleted` because all coroutines will be completed after `stop()`.

Reply:
added comments to the test

Reply:
@korshaknn how about:
```
tripSession.start()
tripSession.route = route
tripSession.route = null
tripSession.stop()
...
verifySequence {
  navigator.setRoute(route)
  navigator.setRoute(null)
}
```

would that work?

Reply:
fixed